core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -135,6 +135,13 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
     val execCoresProp = Option(context.getLocalProperty(EXECUTOR_CORES_LOCAL_PROPERTY))
     val memoryMb = Option(context.getLocalProperty(PYSPARK_MEMORY_LOCAL_PROPERTY)).map(_.toLong)
     val localdir = env.blockManager.diskBlockManager.localDirs.map(f => f.getPath()).mkString(",")
+    // if OMP_NUM_THREADS is not explicitly set, override it with the number of cores
+    if (conf.getOption("spark.executorEnv.OMP_NUM_THREADS").isEmpty) {
+      // SPARK-28843: limit the OpenMP thread pool to the number of cores assigned to this executor
+      // this avoids high memory consumption with pandas/numpy because of a large OpenMP thread pool
+      // see https://github.com/numpy/numpy/issues/10455
+      execCoresProp.foreach(envVars.put("OMP_NUM_THREADS", _))
```

Comment:
We should get this from `spark.task.cpus` or something equivalent from resource profile so the current tasks can uses the number of threads allocated for the number of cores to use (`spark.task.cpus`).

Since this is not enabled by default, I am fine restoring this but would be great if we can track and handle them. Otherwise SPARK-41188 persists with resource profile enabled.

Reply:
Thanks for the clarification! Make sense.

Reply:
Created https://issues.apache.org/jira/browse/SPARK-42613

Reply:
Will create an PR once this is merged